tools/eslint-rules/alphabetize-errors.js

```diff
@@ -19,16 +17,35 @@ function isDefiningError(node) {
 
 module.exports = {
   create: function(context) {
-    var previousNode;
-
+    let previousNode;
     return {
       ExpressionStatement: function(node) {
-        if (isDefiningError(node)) {
-          if (previousNode && !isAlphabetized(previousNode, node)) {
-            context.report({ node: node, message: message });
-          }
-
+        if (!isDefiningError(node)) return;
+        if (!previousNode) {
           previousNode = node;
+          return;
+        }
+        const prev = errorForNode(previousNode);
+        const curr = errorForNode(node);
+        previousNode = node;
+        if (prev >= curr) {
+          let i;
+          for (i = 0; i < Math.min(prev.length, curr.length); ++i)
+            if (prev[i] > curr[i]) break;
+          const message = [prefix, prev, opStr, curr].join('');
+          const j = message.indexOf(opStr);
+          const k = prefix.length + i;
+          const l = j + oLen + i;
+          const kc = message[k];
+          const lc = message[l];
+          const km = `(${kc.charCodeAt(0)})${kc}`;
+          const lm = `${lc}(${lc.charCodeAt(0)})`;
+          let tmp = ' '.repeat(j) + opStr + ' '.repeat(l - oLen - j);
+          const k2 = k - lm.length + 1;
+          const k3 = k2 + km.length;
+          const message2 = tmp.substr(0, k2) + km + tmp.substr(k3) + lm;
```

Comment:
Isn't it easier to just show the index where the two strings differ?

Reply:
Easy? 😤 easy for someone else...
<sub>yeah I was bored this morning</sub>

But seriously, I think it's a nice and informative feedback, and I don't assume this error should happen too often, so I can take it or leave it..

Reply:
My point is that I don't care if `_ >= O` for example. I find it sufficient to know the index where the issue happens (e.g. 7 for `ERR_OUT_OF_RANGE` and `ERR_OUTOFMEMORY`) but I understand this is very personal.

Reply:
Ack.